### PR TITLE
[rv_dm,dv] Use the "right enable" in *_jtag_dmi_debug_disabled_vseq

### PR DESCRIPTION
**Pull request by rswarbrick**
This PR looks reasonably large because it depends on #23799. Once that has been merged, there are two remaining commits:
- *[rv_dm,dv] Make DMI bringup in base vseq depend on pinmux connection*
- *[rv_dm,dv] Use the "right enable" in \*_jtag_dmi_debug_disabled_vseq*

The first commit is a tidyup in the base vseq (which I noticed when trying to fix the vseq for the second commit). The second commit is the main point of this PR. Its commit message is as follows:

*[rv_dm,dv] Use the "right enable" in \*_jtag_dmi_debug_disabled_vseq*

The code here was a bit broken and I clearly hadn't previously
understood the difference between "pinmux enabled" and "debug
enabled".

The *whole point* of this test is supposed to be that DMI transactions
get blocked when pinmux is not enabled. After the changes here, we
actually test that property!

### CODE REVIEW

**Review by vogelpi — APPROVED** [review]
This looks good as well, thanks @rswarbrick .

**Comment by rswarbrick** [comment]
Thanks. I'm going to force-push after rebasing (because a bunch of CI(private) tests failed last time for unrelated IT issues. Assuming things look reasonable, I intend to merge this when they finish. (But it's an evening with children! If someone else gets here first, please feel free to hit the merge button! :-D)

**Pull request merged**
